cloud-sql-java-functions-benchmark/src/main/java/functions/Main.java

```diff
@@ -0,0 +1,185 @@
+    connProps.setProperty("cloudSqlInstance", CONNECTION_NAME);
+
+    LOGGER.log(Level.INFO, "Started logging.");
+
+    LOGGER.log(Level.INFO, "Started logging.");
+    // Create thread pool
+    ScheduledExecutorService execService = Executors.newScheduledThreadPool(2);
+
+    // Start tests for regular connections
+    execService.scheduleAtFixedRate(() -> {
+      connectRegular(jdbcURL, connProps);
+    }, 0, interval, TimeUnit.MILLISECONDS);
+
+    // Start tests for pooled connections
+    DataSource pool = createPool(jdbcURL, connProps);
+    execService.scheduleAtFixedRate(() -> {
+      connectWithPool(pool);
+    }, 0, interval, TimeUnit.MILLISECONDS);
+
+    try {
+      TimeUnit.MILLISECONDS.sleep(duration);
+    } catch (InterruptedException e) {
+      response.setStatusCode(HttpURLConnection.HTTP_INTERNAL_ERROR);
+      BufferedWriter writer = response.getWriter();
+      writer.write(ExceptionUtils.getStackTrace(e));
+      return;
+    }
+
+    execService.shutdown();
+
+    BufferedWriter writer = response.getWriter();
+    writer.write(
+        String.format("Logged connection attempts to Stackdriver Trace for %d ms", duration));
+  }
+
+  private static void connectRegular(String jdbcURL, Properties connProps) {
+    try (Scope rootSpan = TRACER.spanBuilder("regular-connection").startScopedSpan()) {
+      Connection conn;
+
+      // Get connection
+      try (Scope getConnSpan = TRACER.spanBuilder("getConnection").startScopedSpan()) {
+        conn = DriverManager.getConnection(jdbcURL, connProps);
+      } catch (Exception ex) {
+        LOGGER.log(Level.WARNING, "[regular-connection] Error occurred during connection.", ex);
+        return;
+      }
+
+      // Test connection
+      executeStatement(conn);
+
+      // Close connection
+      closeConnection(conn);
+    }
+    LOGGER.log(Level.INFO, "[regular-connection] complete.");
+  }
+
+  private static void connectWithPool(DataSource pool) {
+    try (Scope rootSpan = TRACER.spanBuilder("pool-connection").startScopedSpan()) {
+      Connection conn;
+
+      // Get connection
+      try (Scope getConnSpan = TRACER.spanBuilder("getConnection").startScopedSpan()) {
+        conn = pool.getConnection();
+      } catch (Exception ex) {
+        LOGGER.log(Level.WARNING, "[pool-connection] Error occurred during connection.", ex);
+        return;
+      }
+
+      // Test connection
+      executeStatement(conn);
+
+      // Close connection
+      closeConnection(conn);
+    }
+    LOGGER.log(Level.INFO, "[pool-connection] complete.");
+  }
+
+  private static DataSource createPool(String jdbcURL, Properties props) {
+    HikariConfig config = new HikariConfig();
+    config.setJdbcUrl(jdbcURL);
+    config.setDataSourceProperties(props);
+
+    config.setMaximumPoolSize(5);
+    config.setMinimumIdle(5);
+    config.setConnectionTimeout(TimeUnit.MILLISECONDS.convert(10, TimeUnit.SECONDS)); // 10 seconds
```

Comment:
nit: I'd maybe consider bumping this to 30 since we think it might take 6-8 secs for the first connection to be established.

Reply:
done